heracles/io.py

```diff
@@ -618,7 +618,10 @@ def __getitem__(self, key):
         data = self._cache.get(ext)
         if data is None:
             with self.fits as fits:
-                data = self.reader(fits[ext])
+                if ext in fits:
+                    data = self.reader(fits[ext])
+                else:
+                    raise KeyError(ext)
```

Comment:
Slightly more idiomatic to not have the assignment be in a branch
```suggestion
                if ext not in fits:
                    raise KeyError(ext)
                data = self.reader(fits[ext])
```